src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPSynchronisationService.kt

```diff
@@ -0,0 +1,138 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.csip
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.csip.CSIPSynchronisationService.MappingResponse.MAPPING_FAILED
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.integration.history.DuplicateErrorResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.listeners.SynchronisationMessageType
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.CSIPMappingDto
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.InternalMessage
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.SynchronisationQueueService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.SynchronisationType
+
+@Service
+class CSIPSynchronisationService(
+  private val nomisApiService: CSIPNomisApiService,
+  private val csipMappingApiService: CSIPMappingService,
+  private val csipService: CSIPService,
+  private val telemetryClient: TelemetryClient,
+  private val queueService: SynchronisationQueueService,
+) {
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  suspend fun synchroniseCSIPInsert(event: CSIPOffenderEvent) {
+    // Should never happen
+    if (event.auditModuleName == "DPS_SYNCHRONISATION") {
+      telemetryClient.trackEvent(
+        "csip-synchronisation-skipped",
+        event.toTelemetryProperties(),
+      )
+      return
+    }
+
+    val nomisCSIP = nomisApiService.getCSIP(event.nomisCSIPId)
+    csipMappingApiService.findNomisCSIPMapping(
+      nomisCSIPId = event.nomisCSIPId,
```

Comment:
very very minor I suppose this should be findCSIPByNomisId

Reply:
yes